modules/tinymce/src/themes/silver/main/ts/ui/core/complex/utils/Select.ts

```diff
@@ -0,0 +1,43 @@
+import { Arr } from '@ephox/katamari';
+
+import { Strings} from '@ephox/katamari';
+
+export interface BasicSelectItem {
+  readonly title: string;
+  readonly format: string;
+  readonly icon?: string;
+}
+
+export enum Delimiter {
+  SemiColon = ';',
+  Space = ' '
+}
+
+const process = (rawFormats: string[]): BasicSelectItem[] => Arr.map(rawFormats, (item) => {
+  let title = item, format = item;
+  // Allow text=value block formats
+  const values = Arr.filter(item.split('='), (value) => Strings.isNotEmpty(value));
```

Comment:
Nicely done - but that's not quite all the edge cases. We do use `Strings` most of the time, but I think what would be best here is to filter out strings that are empty or contain only whitespace. Maybe someone makes a typo like `Heading = = h1;` (that's probably a good test to add, actually).

Code to check whitespace is in a different module for.... reasons.
https://github.com/tinymce/tinymce/blob/b4c1a77689f2cd7cdd65bd4456f7de405430f658/modules/tinymce/src/core/main/ts/text/Whitespace.ts#L5